examples/winter_village_train.js

```diff
@@ -0,0 +1,60 @@
+/*
+ *
+ * This runs the train under our Christmas tree. It uses the 10254 Winter Holiday Train retrofitted with a Powered UP hub and train motor.
+ * It also uses a WeDo 2.0 hub with Powered UP distance sensor to detect the train approaching the station and slow it down.
+ * 
+ * Note that if you want to use this yourself you don't need to use a WeDo 2.0 hub, you can use any hub that can accept a distance or color/distance sensor.
+ * 
+ * The maximum speed of the train is set to a constant 50. A further improvement can be made by scaling the speed up according to the current battery voltage,
+ * so the speed doesn't slow as the battery starts dying.
+ *
+ */
+
+const PoweredUP = require("..");
+
+const poweredUP = new PoweredUP.PoweredUP();
+poweredUP.scan(); // Start scanning for hubs
+
+console.log("Looking for Hubs...");
+
+let train = null;
+let sensor = null;
+
+let ramping = false;
+
+poweredUP.on("discover", async (hub) => { // Wait to discover hubs
+
+    if (hub.name === "NK_Winter_Train") {
+        await hub.connect(); // Connect to hub
+        console.log(`Connected to train!`);
+        train = hub;
+    } else if (hub.name === "NK_Winter_Sensor") {
+        await hub.connect(); // Connect to hub
+        console.log(`Connected to sensor!`);
+        sensor = hub;
+
+        sensor.on("distance", (_, distance) => {
+            if (distance < 5 && !ramping) {
+                await stopTrain();
+            }
+        });
```

Comment:
This listen a device (sensor) event on the hub, does it mean we support device events at the hub level ?

Reply:
Oops, I shouldn’t have committed that! :) It’s for my house - designed to run on the old code (master). But I haven’t though more about supporting that as of now.

Reply:
It makes a nice example !

Reply:
the new `waitForDeviceByType` should allow you to migrate this to the new paradigm

Reply:
Yeah - it was an idea that only occurred to me last night while thinking about other stuff, but it solves a problem that many people have, ie. not knowing when to call `getDeviceByType` without watching for an `attach` event, by which point it's provided to you anyway. It just makes it easier to maintain a more sequential coding pattern, which is easier for a lot of people.

Reply:
It's also how I was this could support internal motors/functions on a hub rather on device. ie. If a boost internal motor has a method `setSpeed(speed)`, I can create a lightweight wrapper on the hub called `setSpeed(port, speed)`, which simply passes through the values after using `waitForDeviceByPort`.

Reply:
Awesome